system/Commands/Utilities/Optimize.php

```diff
@@ -99,32 +115,73 @@ private function removeFile(string $cache): void
-            ]
-        );
+        // Prepare search and replace mappings
+        $searchReplace = [];
+
+        if ($restoreDefaults === true) {
+            $searchReplace = [
+                'public bool $configCacheEnabled = true;'  => 'public bool $configCacheEnabled = false;',
+                'public bool $locatorCacheEnabled = true;' => 'public bool $locatorCacheEnabled = false;',
+            ];
+        } else {
+            if ($enableConfigCache === true) {
+                $searchReplace['public bool $configCacheEnabled = false;'] = 'public bool $configCacheEnabled = true;';
+            }
 
-        if ($result) {
-            CLI::write(
-                'Config Caching and FileLocator Caching are enabled in "app/Config/Optimize.php".',
-                'green'
-            );
+            if ($enableLocatorCache === true) {
+                $searchReplace['public bool $locatorCacheEnabled = false;'] = 'public bool $locatorCacheEnabled = true;';
+            }
 
-            return;
+            // If no options provided, update both
+            if ($enableConfigCache === null && $enableLocatorCache === null) {
+                $searchReplace = [
+                    'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
+                    'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
+                ];
+            }
         }
 
-        CLI::error('Error in updating file: ' . clean_path($config));
+        // Apply replacements if necessary
+        if ($searchReplace !== []) {
+            $result = $publisher->replace($config, $searchReplace);
 
-        throw new RuntimeException(__METHOD__);
+            if ($result === true) {
+                $messages = [];
+
+                if (in_array('public bool $configCacheEnabled = true;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $locatorCacheEnabled = true;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is enabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $configCacheEnabled = false;', $searchReplace, true)) {
+                    $messages[] = 'Config Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                if (in_array('public bool $locatorCacheEnabled = false;', $searchReplace, true)) {
+                    $messages[] = 'FileLocator Caching is disabled in "app/Config/Optimize.php".';
+                }
+
+                CLI::write(implode("\n\n", $messages), 'green');
+                CLI::write();
+
+                return;
+            }
+
+            CLI::error('Error in updating file: ' . clean_path($config));
+
+            throw new RuntimeException(__METHOD__);
```

Comment:
We don't throw RuntimeException. Use `CodeIgniter\Exceptions\RuntimeException` instead.